mainsite/catalog/models.py

```diff
@@ -82,7 +85,26 @@ def get_preferred_first_name(self):
     item_description = models.CharField(max_length=1500, blank=True)
 
     #Picture that is uploaded on item creation
-    item_picture = models.ImageField(upload_to='catalog/%Y/%m/%d/', height_field=None, width_field=None)
+    # now represented by separate models with a foreignkey relation
+    #  item_picture = models.ImageField(upload_to='catalog/%Y/%m/%d/', height_field=None, width_field=None)
+    @property
+    def item_picture(self):
+        if self.pictures.exists():
+            return self.pictures.first().picture
+        else:
+            return None
+            #SimpleUploadedFile(name="imagenotfound.jpg",
+                #content=open("/static/mainsite/images/imagenotfound.jpg", 'rb').read(), content_type='image/jpeg')
+    '''@item_picture.setter
+    def item_picture(self, value):
+        if self.pictures.exists():
+            print("1")
+            for pic in self.pictures.all():
+                print(pic)
+            print("2")
+            raise FieldError("Temporary for now!")
+        else:'''
+    #CatalogItemPicture.objects.create(picture=value, item=self, position=1 )
```

Comment:
All this commented out code should probably be removed.

Reply:
✔️